warehouse/warehouse.go

```diff
@@ -1023,6 +1046,8 @@ func (wh *HandleT) initWorkers() {
 							warehouseutils.DestStat(stats.CountType, "failed_uploads", job.Warehouse.Destination.ID).Count(1)
 							break
 						}
+						burstRetryCache(job.Warehouse)
```

Comment:
There is no other reference to cache in this file.
So, instead of directly calling here, can we do something like this in scheduling.go

```
func onSuccessfulUpload(wh wareHouse ) {
  delete(cache, connString(wh))
}
```

call onSuccessfulUpload(job.Warehouse)